### PR TITLE
[CI:DOCS] Spec proposal

### PR DESCRIPTION
**Pull request by baude**
warning: the naming of this might change as well as the location.
    
this is a build on a PR from mheon from last year that proposes a shift from our current approach of creating containers based on the arbitrarily made createconfig.  the new approach would be to have a specification that is detached from the podman cli.  the spec could then be generated and used to make a container.  this theoretically is the beginning of a long-needed refactor involving how we get from the cli -> libpod | apiv2 -> libpod with code re-use and less duplication.
    
the intent is to build the apiv2 container creation based on this approach only.  wiring to the podman cli will happen after the fact.

### CODE REVIEW

File: pkg/specgen/specgen.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+	// OOMScoreAdj adjusts the score used by the OOM killer to determine
+	// processes to kill for the container's process.
+	// Optional.
+	OOMScoreAdj *int `json:"oom_score_adj,omitempty"`
+}
+
+// ContainerHealthCheckConfig describes a container healthcheck with attributes
+// like command, retries, interval, start period, and timeout.
+type ContainerHealthCheckConfig struct {
+	manifest.Schema2HealthConfig `json:"healthcheck,omitempty"`
```
Comment (mheon) [inline pkg/specgen/specgen.go]: Do we want to give this a name? In case we add extra fields related to healthchecks at some point?

Comment (baude) [reply pkg/specgen/specgen.go]: @mheon i couldnt think of a name that wasnt redundant.  you?

Comment (mheon) [reply pkg/specgen/specgen.go]: ...HealthConfig? Maybe?

**Comment by baude** [comment]
updated

**Comment by mheon** [comment]
Current version LGTM, we can discuss further details as the implementation comes together

**Pull request merged**